pkg/client/typed/discovery/discovery_client.go

```diff
@@ -163,6 +171,50 @@ func (d *DiscoveryClient) ServerResources() (map[string]*unversioned.APIResource
 	return result, nil
 }
 
+// serverPreferredResources returns the supported resources with the version preferred by the
+// server. If namespaced is true, only namespaced resources will be returned.
+func (d *DiscoveryClient) serverPreferredResources(namespaced bool) ([]unversioned.GroupVersionResource, error) {
```

Comment:
So does the garbage collector ignore the version specified in the owner reference? Wondering how that will work with third party resource data objects whose versions are independent and unconnected.